### PR TITLE
Nxos become

### PR DESCRIPTION
**Pull request by acozine**
##### SUMMARY
Updates network documentation to cover changes in https://github.com/ansible/ansible/pull/39355

##### ISSUE TYPE
 - Docs Pull Request

##### COMPONENT NAME
docs.ansible.com

##### ANSIBLE VERSION
2.5
Will backport to `stable-2.5`

### CODE REVIEW

**Review by gundalow — COMMENTED** [review]
1) Do we want people to use `authorize`
2) This functionality is only in 2.5.3 (?) onwards. That needs to be clearer in the docs

@trishnaguha How come this is getting backported, Is it counted as a bug fix, rather than feature work?

File: docs/docsite/rst/network/user_guide/platform_nxos.rst [hunk 1]
```diff
@@ -25,8 +25,9 @@ Connections Available
 | |                         | |                                             | | Requires ``transport: nxapi``         |
 | |                         | |                                             | | in the ``provider`` dictionary        |
 +---------------------------+-----------------------------------------------+-----------------------------------------+
-| | **Enable Mode**         | | not supported by NXOS                       | | not supported by NXOS                 |
-| | (Privilege Escalation)  | |                                             |                                         |
+| | **Enable Mode**         | | supported - use ``ansible_become: yes``     | | supported - use ``authorize: yes``    |
```
Comment (gundalow) [inline docs/docsite/rst/network/user_guide/platform_nxos.rst]: Note about 2.5.3 here?

Comment (acozine) [reply docs/docsite/rst/network/user_guide/platform_nxos.rst]: addressed by 119f45e286d59b54916965f7ce0aaec86bac57cf?

Comment (gundalow) [reply docs/docsite/rst/network/user_guide/platform_nxos.rst]: Looks good

**Comment by trishnaguha** [comment]
@gundalow bugfix

**Comment by trishnaguha** [comment]
@gundalow @acozine `authorize`, `auth_pass` can be used with connecion: local transport cli. since we now support ansible-ish way for privilege escalation since 2.5, the doc looks good to me.
Just FYI nxapi code for enable mode isn't pushed yet. So the doc for nxapi needs to wait for now.

File: docs/docsite/rst/network/user_guide/platform_nxos.rst [hunk 2]
```diff
@@ -96,6 +100,8 @@ Example NX-API ``group_vars/nxos.yml``
    nxapi:
      host: "{{ inventory_hostname }}"
      transport: nxapi
+     authorize: yes
+     auth_pass: !vault...
```
Comment (trishnaguha) [inline docs/docsite/rst/network/user_guide/platform_nxos.rst]: this also needs to be removed from nxapi provider spec.

**Comment by acozine** [comment]
Per discussion with the cisco nxos team, nxapi itself doesn't support enable mode - reflected in the docs.

**Review by trishnaguha — APPROVED** [review]

**Review by gundalow — APPROVED** [review]

**Comment by gundalow** [comment]
Please backport into `stable-2.5`

**Comment by trishnaguha** [comment]
cherry-picked to 2.5 https://github.com/ansible/ansible/pull/40221

**Pull request merged**